python/tank/commands/core_localize.py

```diff
@@ -77,26 +78,68 @@ def run_interactive(self, log, args):
 
         pc_root = self.tk.pipeline_configuration.get_path()
         log.debug("Executing the localize command for %r" % self.tk)
-        return do_localize(log, pc_root, suppress_prompts=False)
+        return do_localize(
+            log,
+            self.tk.shotgun,
+            pc_root,
+            suppress_prompts=False
+        )
 
 
-def do_localize(log, pc_root_path, suppress_prompts):
+def do_localize(log, sg_connection, target_config_path, suppress_prompts):
     """
     Perform the actual localize command.
 
     :param log: logging object
-    :param pc_root_path: Path to the config that should be localized.
-    :param suppress_prompts: Boolean to indicate if no questions should be asked.
+    :param sg_connection: An open shotgun connection
+    :param str target_config_path: Path to the config that should be localized.
+    :param bool suppress_prompts: Indicates if prompts should be suppressed.
     """
-    pipeline_config = pipelineconfig_factory.from_path(pc_root_path)
+
+    # the configuration to localize
+    target_pipeline_config = pipelineconfig_factory.from_path(
+        target_config_path)
+
+    # the core install location for the current config. this will be resolved to
+    # a linked config or the config where the running core lives
+    source_config_path = target_pipeline_config.get_install_location()
 
     log.info("")
-    if pipeline_config.is_localized():
-        raise TankError("Looks like your current pipeline configuration already has a local install of the core!")
+    if target_pipeline_config.is_localized():
+        # if we're here, there's already a core in the config's install folder
+        raise TankError(
+            "Looks like your current pipeline configuration already has a "
+            "local install of the core!"
+        )
 
-    core_api_root = pipeline_config.get_install_location()
+    # if a core descriptor is supplied, ensure it is cached locally and use it
+    # as the core to localize.
+    if pipelineconfig_utils.has_core_descriptor(target_config_path):
+        core_descriptor = pipelineconfig_utils.get_core_descriptor(
+            target_config_path,
+            sg_connection
+        )
+        core_descriptor.ensure_local()
+        source_core_path = core_descriptor.get_path()
+        source_core_version = core_descriptor.get_version()
+    else:
+        # fall back to using the core that exists in the source config
+        source_core_path = os.path.join(
+            source_config_path,
+            "install",
+            "core"
+        )
+
+        # resolve the version of core
+        source_core_version = \
+            target_pipeline_config.get_associated_core_version()
+
+    log.info(
```

Comment:
thinking maybe for clarity, better to have two different messages here, one for the old case when we copy the core across and one for the new core_api.yml case - maybe rather than using the verb copy it could say something like "Installing required core v0.18.120 into pipeline config /foo/bar"?